src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -402,6 +422,91 @@ private static void ThrowInvalidDocumentId(string id)
             return false;
         }
 
+        private void RecreateCounters(DocumentsOperationContext context, string id, BlittableJsonReaderObject document,
+            BlittableJsonReaderObject metadata, ref DocumentFlags flags)
+        {
+
+            var countersStorage = context.DocumentDatabase.DocumentsStorage.CountersStorage;
+            var onDiskCounters = countersStorage.GetCountersForDocument(context, id, 0, int.MaxValue);
+            var sortedCounters = new HashSet<string>(onDiskCounters, StringComparer.OrdinalIgnoreCase).ToList();
```

Comment:
Don't need the `HashSet` here. 
Also, pretty sure that we already get them sorted at this point.